### ISSUE TITLE
Partial Refund Causing Reporting Inaccuracies

### ISSUE THREAD

**Issue by tagtaylor** [woocommerce/woocommerce#18401]
When processing refunds inside a particular order, we charge for the item and flat rate shipping.  If we process a refund, we refund only the line item for the product but not the shipping.  The total amount refunded will then show in reports as a part of total refunds, but the number of orders refunded stays the same unless we fully refund the entire order.  Alternatively, if we process the refund of the product, and then change the status of the order to "refunded" it then proceeds to deduct the remaining shipping charge and show it as fully refunded, though no refund for this line item has been processed.

## WordPress Environment

<details>
```
`
### WordPress Environment ###

Home URL: n/a
Site URL: n/a
WC Version: 3.2.6
Log Directory Writable: ✔
WP Version: 4.9.1
WP Multisite: –
WP Memory Limit: 768 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US

### Server Environment ###

Server Info: Apache
PHP Version: 5.6.32
PHP Post Max Size: 128 MB
PHP Time Limit: 0
PHP Max Input Vars: 3000
cURL Version: 7.53.0
OpenSSL/1.0.2k

SUHOSIN Installed: –
MySQL Version: 5.6.36
Max Upload Size: 128 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 3.2.6
WC Database Prefix: wp_
MaxMind GeoIP Database: ✔
Total Database Size: 179.51MB
Database Data Size: 130.77MB
Database Index Size: 48.74MB
wp_woocommerce_sessions: Data: 18.58MB + Index: 1.52MB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.05MB
wp_woocommerce_order_items: Data: 1.52MB + Index: 0.33MB
wp_woocommerce_order_itemmeta: Data: 5.52MB + Index: 6.03MB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.05MB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.05MB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB
wp_aepc_custom_audiences: Data: 0.02MB + Index: 0.00MB
wp_amz_assets: Data: 0.00MB + Index: 0.00MB
wp_amz_cross_sell: Data: 0.00MB + Index: 0.00MB
wp_amz_products: Data: 0.00MB + Index: 0.00MB
wp_amz_queue: Data: 0.00MB + Index: 0.00MB
wp_amz_report_log: Data: 0.12MB + Index: 0.01MB
wp_amz_search: Data: 0.00MB + Index: 0.00MB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB
wp_comments: Data: 3.52MB + Index: 1.98MB
wp_duplicator_packages: Data: 0.00MB + Index: 0.00MB
wp_edn_subscriber: Data: 0.00MB + Index: 0.00MB
wp_ghostmonitor_data: Data: 14.56MB + Index: 0.00MB
wp_huge_it_videogallery_galleries: Data: 0.00MB + Index: 0.00MB
wp_huge_it_videogallery_videos: Data: 0.01MB + Index: 0.00MB
wp_links: Data: 0.02MB + Index: 0.02MB
wp_nf3_actions: Data: 0.00MB + Index: 0.00MB
wp_nf3_action_meta: Data: 0.03MB + Index: 0.01MB
wp_nf3_fields: Data: 0.01MB + Index: 0.00MB
wp_nf3_field_meta: Data: 0.22MB + Index: 0.03MB
wp_nf3_forms: Data: 0.00MB + Index: 0.00MB
wp_nf3_form_meta: Data: 0.02MB + Index: 0.01MB
wp_nf3_objects: Data: 0.00MB + Index: 0.00MB
wp_nf3_object_meta: Data: 0.01MB + Index: 0.00MB
wp_nf3_relationships: Data: 0.00MB + Index: 0.00MB
wp_nxs_log: Data: 0.03MB + Index: 0.01MB
wp_options: Data: 3.52MB + Index: 0.13MB
wp_postmeta: Data: 22.55MB + Index: 21.06MB
wp_posts: Data: 7.52MB + Index: 1.06MB
wp_qb_bars: Data: 0.00MB + Index: 0.00MB
wp_qb_conversions: Data: 0.09MB + Index: 0.08MB
wp_qb_views: Data: 3.18MB + Index: 2.56MB
wp_revslider_css: Data: 0.13MB + Index: 0.00MB
wp_revslider_layer_animations: Data: 0.02MB + Index: 0.00MB
wp_revslider_navigations: Data: 0.02MB + Index: 0.00MB
wp_revslider_sliders: Data: 0.20MB + Index: 0.00MB
wp_revslider_slides: Data: 1.52MB + Index: 0.00MB
wp_revslider_static_slides: Data: 0.02MB + Index: 0.00MB
wp_rg_form: Data: 0.00MB + Index: 0.00MB
wp_rg_form_meta: Data: 0.00MB + Index: 0.00MB
wp_rg_form_view: Data: 0.00MB + Index: 0.00MB
wp_rg_incomplete_submissions: Data: 0.00MB + Index: 0.00MB
wp_rg_lead: Data: 0.00MB + Index: 0.00MB
wp_rg_lead_detail: Data: 0.00MB + Index: 0.01MB
wp_rg_lead_detail_long: Data: 0.00MB + Index: 0.00MB
wp_rg_lead_meta: Data: 0.00MB + Index: 0.00MB
wp_rg_lead_notes: Data: 0.00MB + Index: 0.00MB
wp_smush_dir_images: Data: 0.02MB + Index: 0.03MB
wp_termmeta: Data: 0.02MB + Index: 0.03MB
wp_terms: Data: 0.19MB + Index: 0.28MB
wp_term_relationships: Data: 0.23MB + Index: 0.23MB
wp_term_taxonomy: Data: 0.20MB + Index: 0.23MB
wp_usermeta: Data: 7.52MB + Index: 10.03MB
wp_users: Data: 0.41MB + Index: 0.42MB
wp_wcch_page_history: Data: 28.56MB + Index: 0.00MB
wp_wfBadLeechers: Data: 0.02MB + Index: 0.00MB
wp_wfBlockedCommentLog: Data: 0.02MB + Index: 0.00MB
wp_wfBlockedIPLog: Data: 0.02MB + Index: 0.00MB
wp_wfBlocks: Data: 0.02MB + Index: 0.02MB
wp_wfBlocksAdv: Data: 0.02MB + Index: 0.00MB
wp_wfConfig: Data: 0.27MB + Index: 0.00MB
wp_wfCrawlers: Data: 0.02MB + Index: 0.00MB
wp_wfFileMods: Data: 2.52MB + Index: 0.00MB
wp_wfHits: Data: 4.52MB + Index: 1.89MB
wp_wfHoover: Data: 0.02MB + Index: 0.02MB
wp_wfIssues: Data: 0.02MB + Index: 0.00MB
wp_wfKnownFileList: Data: 1.52MB + Index: 0.00MB
wp_wfLeechers: Data: 0.48MB + Index: 0.00MB
wp_wfLockedOut: Data: 0.02MB + Index: 0.00MB
wp_wfLocs: Data: 0.02MB + Index: 0.00MB
wp_wfLogins: Data: 0.06MB + Index: 0.03MB
wp_wfNet404s: Data: 0.02MB + Index: 0.02MB
wp_wfNotifications: Data: 0.02MB + Index: 0.00MB
wp_wfPendingIssues: Data: 0.02MB + Index: 0.00MB
wp_wfReverseCache: Data: 0.02MB + Index: 0.00MB
wp_wfScanners: Data: 0.27MB + Index: 0.00MB
wp_wfSNIPCache: Data: 0.02MB + Index: 0.05MB
wp_wfStatus: Data: 0.19MB + Index: 0.13MB
wp_wfThrottleLog: Data: 0.02MB + Index: 0.02MB
wp_wfVulnScanners: Data: 0.02MB + Index: 0.00MB
wp_wpeditor_settings: Data: 0.00MB + Index: 0.00MB
wp_wpie_export_log: Data: 0.00MB + Index: 0.00MB
wp_wpmcleaner: Data: 0.02MB + Index: 0.00MB
wp_yith_wcwl: Data: 0.02MB + Index: 0.02MB
wp_yith_wcwl_lists: Data: 0.02MB + Index: 0.03MB
wp_yoast_seo_links: Data: 0.09MB + Index: 0.05MB
wp_yoast_seo_meta: Data: 0.06MB + Index: 0.00MB
wp_ywcars_messagemeta: Data: 0.02MB + Index: 0.00MB
wp_ywcars_messages: Data: 0.02MB + Index: 0.00MB

### Post Type Counts ###

attachment: 447
custom_css: 1
custom-css-js: 3
customize_changeset: 3
emd_video: 4
etheme_portfolio: 6
nav_menu_item: 8
nf_sub: 370
oembed_cache: 13
page: 63
post: 211
product: 3
product_variation: 37
revision: 648
sg_optimizer_jobs: 62
shop_coupon: 55
shop_order: 4216
shop_order_refund: 399
shop_webhook: 1
staticblocks: 6
testimonials: 4
vc_grid_item: 1
wpsl_stores: 14

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (36) ###

Gravity Forms: by rocketgenius – 2.2.1
Redux Framework: by Team Redux – 3.6.7.7
WooCommerce added to cart popup (Ajax): by XootiX – 1.3 – Not tested with the active version of WooCommerce
All-in-One WP Migration: by ServMask – 6.61
Cresta Social Share Counter: by CrestaProject - Rizzo Andrea – 2.7.6
Simple Custom CSS and JS: by Diana Burduja – 3.12
XStore Core: by 8theme – 1.0.19
jQuery in Posts, Pages, Custom Post Types: by WebCodingPlace – 4.0
WPBakery Page Builder: by Michael M - WPBakery.com – 5.4.5
MailChimp for WordPress: by ibericode – 4.1.13
Ninja Forms: by The WP Ninjas – 3.2.8
Olark Live Chat: by Olark – 1.0.8
Pixel Caffeine: by AdEspresso – 2.0.0
Quick Page/Post Redirect Plugin: by anadnet – 5.1.8
SG Optimizer: by SiteGround – 3.3.5
Stamped.io Product Reviews: by Stamped.io – 1.0
Revive Old Post (Former Tweet Old Post): by revive.social – 7.5.0
WooCommerce Checkout Field Editor: by WooCommerce – 1.5.9
WooCommerce Customer History: by Brian Richards – 1.2.1 – Not tested with the active version of WooCommerce
WooCommerce Expand Tabs: by Diana Burduja – 1.7
WooCommerce PayPal Express Checkout Gateway: by WooCommerce – 1.5.0
WooCommerce PayPal Pro (Classic and PayFlow Editions) Gateway: by WooThemes – 4.4.2 – 4.4.11 is available – Not tested with the active version of WooCommerce
WooCommerce Google Analytics Pro: by SkyVerge – 1.3.6
WooCommerce Product Stock Alert: by WC Marketplace
The Grey Parrots – 1.4.7 – Not tested with the active version of WooCommerce

WooCommerce Shipment Tracking: by WooCommerce – 1.6.9
WooCommerce - ShipStation Integration: by WooCommerce – 4.1.19
WooCommerce Social Login: by SkyVerge – 2.3.3
WooCommerce Tab Manager: by SkyVerge – 1.8.4
WooCommerce: by Automattic – 3.2.6
Wordfence Security: by Wordfence – 6.3.22
WordPress Importer: by wordpressdotorg – 0.6.3
Yoast SEO Premium: by Team Yoast – 6.0
WP Rocket: by WP Media – 2.11.3
WP Smush: by WPMU DEV – 2.7.6
WP Store Locator: by Tijmen Smit – 2.2.10
YITH Advanced Refund System for WooCommerce: by YITHEMES – 1.0.1 – Not tested with the active version of WooCommerce

### Settings ###

API Enabled: ✔
Force SSL: ✔
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)


### WC Pages ###

Shop base: #27 - /shop/
Cart: #28 - /cart/
Checkout: #29 - /checkout/
My account: #30 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: XStore Child
Version: 0.0.1
Author URL: http://8theme.com
Child Theme: ✔
Parent Theme Name: XStore
Parent Theme Version: 4.13
Parent Theme Author URL: http://www.8theme.com
WooCommerce Support: ✔

### Templates ###

Overrides: xstore/woocommerce/archive-product.php
xstore/woocommerce/cart/cart-empty.php
xstore/woocommerce/cart/cart-item-data.php
xstore/woocommerce/cart/cart.php
xstore/woocommerce/cart/cross-sells.php
xstore/woocommerce/cart/mini-cart.php
xstore/woocommerce/checkout/form-billing.php
xstore/woocommerce/checkout/form-checkout.php
xstore/woocommerce/checkout/form-shipping.php
xstore/woocommerce/content-product.php
xstore/woocommerce/content-product_cat.php
xstore/woocommerce/content-single-product.php
xstore/woocommerce/content-widget-product.php
xstore/woocommerce/global/breadcrumb.php
xstore/woocommerce/global/form-login.php
xstore/woocommerce/global/quantity-input.php
xstore/woocommerce/global/wrapper-end.php
xstore/woocommerce/global/wrapper-start.php
xstore/woocommerce/loop/loop-end.php
xstore/woocommerce/loop/loop-start.php
xstore/woocommerce/loop/no-products-found.php
xstore/woocommerce/loop/pagination.php
xstore/woocommerce/loop/sale-flash.php
xstore/woocommerce/single-product/add-to-cart/simple.php
xstore/woocommerce/single-product/product-image.php
xstore/woocommerce/single-product/related.php
xstore/woocommerce/single-product/sale-flash.php
xstore/woocommerce/single-product/short-description.php
xstore/woocommerce/single-product/tabs/additional-information.php
xstore/woocommerce/single-product/tabs/description.php
xstore/woocommerce/single-product/tabs/tabs.php
xstore/woocommerce/single-product/title.php
xstore/woocommerce/single-product/up-sells.php
xstore/woocommerce/single-product.php

`
```
</details>

**Comment by mikejolley** [comment]
> but the number of orders refunded stays the same

tl;dr An order with a partial refund is not considered a refunded order.

The order refund count in the reports section only counts fully refunded orders. If you're not refunding all items (shipping included), the count will not update. The 'items refunded' count will though. This appears intentional.

Thanks
